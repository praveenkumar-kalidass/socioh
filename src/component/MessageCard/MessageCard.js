import React from 'react';
import PropTypes from 'prop-types';
import { Body, Card, CardItem, Left, Text, Thumbnail } from 'native-base';

import {
  PrimaryText,
  NotificationContainer,
  Notification,
  Count,
} from './MessageCard.style';

const MessageCard = ({
  image,
  primaryText,
  secondaryText,
  notificationText,
}) => {
  return (
    <>
      <Card>
        <CardItem>
          <Left>
            <Thumbnail large source={image} />
            <Body>
              <PrimaryText testID="card_primary_text">
                {primaryText}
              </PrimaryText>
              <If condition={Boolean(secondaryText)}>
                <Text note testID="card_secondary_text">
                  {secondaryText}
                </Text>
              </If>
            </Body>
            <If condition={Boolean(notificationText)}>
              <NotificationContainer>
                <Notification>
                  <Count testID="card_notification">{notificationText}</Count>
                </Notification>
              </NotificationContainer>
            </If>
          </Left>
        </CardItem>
      </Card>
    </>
  );
};

MessageCard.propTypes = {
  image: PropTypes.any.isRequired,
  primaryText: PropTypes.string.isRequired,
  secondaryText: PropTypes.string,
  notificationText: PropTypes.string,
};

MessageCard.defaultProps = {
  secondaryText: null,
  notificationText: null,
};

export default MessageCard;
