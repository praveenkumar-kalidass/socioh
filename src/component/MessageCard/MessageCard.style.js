import styled from 'styled-components/native';
import { Badge, Text } from 'native-base';

import { COLOR, FONT, SPACING } from '../../constant';

export const PrimaryText = styled(Text)`
  font-size: ${FONT.SIZE_20};
  margin-bottom: ${SPACING.SPACE_10};
`;

export const NotificationContainer = styled.View`
  flex-direction: row;
  justify-content: flex-end;
  align-content: flex-start;
  align-self: stretch;
`;

export const Notification = styled(Badge)`
  background-color: ${COLOR.SECONDARY};
`;

export const Count = styled(Text)`
  color: ${COLOR.BLACK};
`;
