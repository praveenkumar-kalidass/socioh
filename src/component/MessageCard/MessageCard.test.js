import React from 'react';
import { render } from '@testing-library/react-native';

import UserAvatar from '../../asset/image/user.png';
import MessageCard from './index';

describe('MessageCard', () => {
  it('should match snapshot', () => {
    const container = render(
      <MessageCard
        image={UserAvatar}
        primaryText="Praveenkumar"
        secondaryText="Hello World!"
        notificationText="10"
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
