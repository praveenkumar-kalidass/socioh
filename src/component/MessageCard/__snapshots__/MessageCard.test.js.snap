// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageCard should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderBottomWidth": 0.5,
      "borderColor": "#ccc",
      "borderLeftWidth": 0.5,
      "borderRadius": 2,
      "borderRightWidth": 0.5,
      "borderTopWidth": 0.5,
      "elevation": 3,
      "flexWrap": "nowrap",
      "marginBottom": 5,
      "marginLeft": 2,
      "marginRight": 2,
      "marginTop": 5,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 1.5,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Image
        large={true}
        source={
          Object {
            "testUri": "../../../src/asset/image/user.png",
          }
        }
        style={
          Object {
            "borderRadius": 40,
            "height": 80,
            "width": 80,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": null,
            "alignSelf": "center",
            "flex": 1,
            "marginLeft": 10,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "fontSize": 20,
                "marginBottom": 10,
              },
            ]
          }
          testID="card_primary_text"
        >
          Praveenkumar
        </Text>
        <Text
          note={true}
          style={
            Object {
              "color": "#808080",
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "400",
              "marginRight": 20,
            }
          }
          testID="card_secondary_text"
        >
          Hello World!
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignContent": "flex-start",
              "alignSelf": "stretch",
              "flexDirection": "row",
              "justifyContent": "flex-end",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "#ED1727",
                "borderRadius": 13.5,
                "height": 27,
                "justifyContent": "center",
                "paddingBottom": 3,
                "paddingLeft": 6,
                "paddingRight": 6,
                "paddingTop": 3,
              },
              Object {
                "backgroundColor": "#feda68",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "lineHeight": 19,
                  "paddingLeft": 3,
                  "paddingRight": 3,
                  "textAlign": "center",
                },
                Object {
                  "color": "#000000",
                },
              ]
            }
            testID="card_notification"
          >
            10
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
